magento/pwa-studio
# PWA-2313: Update StoreConfig.id GraphQL field to store_code

- Updated StoreConfig.id gql to store_code
- Updated tests



## Description

Overview
The field StoreConfig.id is deprecated. We should use the `store_code` field instead.

By default, Apollo client uses the `__typename` and `id` field for caching so we will also need to update the fields that Apollo client uses to cache StoreConfig objects from `id` to `store_code`. 

Found in:

peregrine/lib/talons/CartPage/ProductListing/product.gql.js
peregrine/lib/talons/CartPage/ProductListing/productListing.gql.ee.js
peregrine/lib/talons/CheckoutPage/ItemsReview/itemsReview.gql.js
peregrine/lib/talons/Cms/cmsPage.gql.js
peregrine/lib/talons/Footer/footer.gql.js
peregrine/lib/talons/Gallery/gallery.gql.ee.js
peregrine/lib/talons/Header/storeSwitcher.gql.js (x2)
peregrine/lib/talons/MiniCart/miniCart.gql.js
peregrine/lib/talons/Navigation/navigation.gql.js
peregrine/lib/talons/OrderHistoryPage/orderHistoryContext.gql.js
peregrine/lib/talons/OrderHistoryPage/orderRow.gql.js
peregrine/lib/talons/ProductFullDetail/productFullDetail.gql.ee.js
peregrine/lib/talons/RootComponents/Product/product.gql.js
peregrine/lib/talons/SearchPage/searchPage.gql.js
peregrine/lib/talons/Wishlist/WishlistDialog/wishlistDialog.gql.js
peregrine/lib/talons/WishlistPage/createWishlist.gql.js
venia-ui/lib/RootComponents/Category/category.gql.js
venia-ui/lib/components/App/localeProvider.js
venia-ui/lib/components/Head/index.js
Acceptance Criteria
PWA Studio no longer requests the `id` field in `StoreConfig` queries
`StoreConfig` objects are still properly cached


## Related Issue







[Closes PWA-2313](https://jira.corp.magento.com/browse/PWA-2313)

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps





Test Plan
Verify search still works correctly, respecting any backend configurations
Verify the main flow (category page > pdp > add to cart > cart > checkout) works as expected, respecting any backend configurations
Verify the wishlist page still works correctly, respecting any backend configurations
Test upgrade flow. So either on cloud or local hit all the changed area above in 12.1 (master)  and then run staging server with this pr and check all areas in new tab of same browser. Details steps on upgrad test can be found here https://jira.corp.magento.com/browse/PWA-1418

#### Test scenario(s) for direct fix/feature





#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
